### PR TITLE
gst-plugins-good: Add optional qt5Support

### PR DESCRIPTION
**Pull request by pstn**
###### Motivation for this change

nheko video calling (Drafted in PR #110595) needs qmlgl, so I put a switch into gst-plugins-good to enable it.
I put another switch in for `lreleaseSupport`, because I don't think it's necessary in most cases but I would be happy to see it when I needed it while working on something.

While working on it, I saw that pulse was disabled twice in the same block, so I also fixed that in another commit.

Thanks @rnhmjoj for the help,
ping @matthewbauer 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review pr 110608"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/gstreamer/good/default.nix [hunk 1]
```diff
@@ -30,6 +30,7 @@
 , mpg123
 , twolame
 , gtkSupport ? false, gtk3 ? null
+, qt5Support ? false, qt5 , lreleaseSupport ? false
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/gstreamer/good/default.nix:33]:
```suggestion
, qt5Support ? false, qt5 , lreleaseSupport
```

File: pkgs/development/libraries/gstreamer/good/default.nix [hunk 2]
```diff
@@ -45,6 +46,7 @@
 
 assert gtkSupport -> gtk3 != null;
 assert raspiCameraSupport -> ((libraspberrypi != null) && stdenv.isLinux && stdenv.isAarch64);
+assert lreleaseSupport -> qt5Support;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/gstreamer/good/default.nix:49]:
```suggestion
```

File: pkgs/development/libraries/gstreamer/good/default.nix [hunk 3]
```diff
@@ -102,6 +104,13 @@ stdenv.mkDerivation rec {
   ] ++ optionals gtkSupport [
     # for gtksink
     gtk3
+  ] ++ optionals qt5Support (with qt5; [
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/gstreamer/good/default.nix:107]: I don't think you want to input qt5 directly but use libforQt5 in callPackages.

File: pkgs/development/libraries/gstreamer/good/default.nix [hunk 4]
```diff
@@ -102,6 +104,13 @@ stdenv.mkDerivation rec {
...
+    qtbase
+    qtdeclarative
+    qtwayland
+    qtx11extras
+  ]) ++ optionals lreleaseSupport [
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/gstreamer/good/default.nix:112]: This option does not exist.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/gstreamer/good/default.nix (hunk 3 above)
Comment (pstn) [reply pkgs/development/libraries/gstreamer/good/default.nix:107]: Are there docs on that?

Comment (rnhmjoj) [reply pkgs/development/libraries/gstreamer/good/default.nix:107]: I'm not sure it's really needed... `libsForQt5.callPackage` ensures all the dependencies are using the same Qt version, which is significant for a Qt application but this is a library and I don't think any other dependency depends on Qt.

**Comment by rnhmjoj** [comment]
So, I've investigated a bit the `lrelease` situation. This is tool needed to compile translation files used by Qt applications.
I looked at the gst-plugins-good [sources](https://github.com/GStreamer/gst-plugins-good/blob/master/ext/qt/qtplugin.pro) and found no mentions of translation (no `TRANSLATIONS` variable in the .pro or .ts files). Since the project builds and works without it I assume the meson message (`Program lrelease found: NO`) is a product of some default template/code generation but it can be ignored.

File: pkgs/development/libraries/gstreamer/good/default.nix (hunk 3 above)
Comment (symphorien) [reply pkgs/development/libraries/gstreamer/good/default.nix:107]: relevant documentation https://nixos.org/manual/nixpkgs/stable/#sec-language-qt

**Comment by rnhmjoj** [comment]
I'd like to get this merged before the other nheko PR. Can you drop the qtrelease thing so we can go on?

**Comment by pstn** [comment]
Whoops, I thought I already did that. Here you go.

**Review by rnhmjoj — APPROVED** [review]
So, enabling `qt5Support` gives a 780M increase to the closure size, so it's definitely better to keep it disabled by default. I also tested that gst-plugins-good and nheko build.

**Pull request merged**